### PR TITLE
Build/lock dependencies

### PR DESCRIPTION
**Pull request by tdeekens**
> This pull request updates the readme and adds lock-files.

### Background & Context

We had some lovely contribution but also some hard time along the way for the contributor due to versions of the dependencies as it seems. This PR tries to bring some clarity to the readme and adds lock-files to enforce reproducible builds.

### Tasks

- Add lock-files for `yarn` and `npm`
- Update readme to point to yarn and npm@5.2
- Update prettier and re-format with new version

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,11 +208,18 @@ All relevant commits will be signed with the key `0x24BB6BF4` for additional sec
 
 ### Development and contributing
 
-We rely on npm-run-scripts for integrating with out tooling infrastructure. We use ESLint as a pre-commit hook to ensure code consistency. Morover, to ease formatting we use [prettier](https://github.com/prettier/prettier) while building the `/dist` assets happens through `rollup`.
+#### Installation (`yarn i`)
 
-These are our npm scripts
+We support both `yarn` and `npm@5.2` officially while providing lock-files for either dependency manager to provide reproducible installs and builds on either or. TravisCI itself is configured to install dependencies using `yarn`. When using an older version of `npm` we can not fully ensure the versions of installed dependencies which might lead to unanticipated problems.
```
Comment (plobbes) [inline README.md]: nit: slightly better wording might be...
--
+We support both `yarn` and `npm@5.2` officially, while providing lock-files for either dependency manager, to provide reproducible installs and builds. TravisCI is configured to install dependencies using `yarn`. When using an older version of `npm` the versions of installed dependencies cannot be insured, which may lead to unanticipated problems.

File: README.md [hunk 2]
```diff
@@ -208,11 +208,18 @@ All relevant commits will be signed with the key `0x24BB6BF4` for additional sec
...
 
-- `npm test` to run our test suite via jsdom and karma
+#### Scripts
+
+We rely on npm run-scripts for integrating with out tooling infrastructure. We use ESLint as a pre-commit hook to ensure code consistency. Moreover, to ease formatting we use [prettier](https://github.com/prettier/prettier) while building the `/dist` assets happens through `rollup`.
```
Comment (plobbes) [inline README.md]: nit: minor tweaks for the first sentence?
--
We rely on `npm run-scripts` for integrating without tooling infrastructure. ...

**Comment by tdeekens** [comment]
Thanks for the feedback. Will incoporate it later.

**Comment by cure53** [comment]
LGTM, thanks all :)

**Comment by tdeekens** [comment]
Ha, so much for taking up the improvements made. 😄

**Pull request merged**